pkgs/development/compilers/ghc/with-packages.nix

```diff
@@ -44,6 +47,7 @@ buildEnv {
       rm -f $out/bin/$prg
       makeWrapper ${ghc}/bin/$prg $out/bin/$prg         \
         --add-flags '"-B$NIX_GHC_LIBDIR"'               \
+        ${optionalString exposeGHC "--add-flags '-package ghc'"} \
```

Comment:
In my opinion, `ghcWithPackages` is not the right place for this feature. This addition feels very awkward. Can you not achieve the same thing with a shell alias or a separate wrapper script? Also, adding `ghc` to the list of dependencies in your project's Cabal file would unhide the package as well. Is that kind of local solution not preferable?